app/controllers/pages_controller.rb

```diff
@@ -9,4 +12,38 @@ def show
       end
     end
   end
+
+  def index
+    @presenter = PagesPresenter.new(pages, search_page, search_params, sort_key, sort_dir, from_date_as_datetime, to_date_as_datetime)
+  end
+
+private
+
+  def pages
+    @pages ||= begin
+      Page.joins(page_visits: [{ visit: [{ survey_visit: [{ survey: [{ survey_answers: [{ mentions: :phrase }] }] }] }] }])
+        .where("surveys.started_at" => date_range)
+        .where("pages.base_path like ?", "%#{q}%")
+        .group("pages.id")
+        .order("#{sort_key} #{sort_dir}")
+        .pluck("pages.base_path as page_base_path", "count(distinct(surveys.id)) as feedback_comments", "pages.id")
+        .map { |page_info| { base_path: page_info.first, survey_count: page_info.second, page_id: page_info.third } }
+    end
+  end
+
+  def sort_key
+    default_key = "feedback_comments"
+    key = params[:sort_key] || default_key
+    %w[page_base_path feedback_comments].include?(key) ? key : default_key
```

Comment:
Hmmm, I see where you're coming from. I like that this means that all the sort_key/sort_dir stuff is in one place, rather than distributed so someone new to the code doesn't end up on a goose chase (admittedly, not like they'll have to go far but I like having things close together/in the same method if possible so they are logically all part of the same thing). It's all subjective though, maybe we'll see what happens as things evolve anyway, if the same sort keys are used all over the app we might end up putting them in a module or something

Reply:
To be fair you already did this for `search_page`. I'm happy with either approach for now.

Reply:
👍